edb/pgsql/compiler/dml.py

```diff
@@ -560,23 +560,14 @@ def process_insert_body(
     values = select.target_list
 
     # The main INSERT query of this statement will always be
-    # present to insert at least the `id` and `__type__`
-    # properties.
+    # present to insert at least the `id` property.
     insert_stmt = insert_cte.query
     assert isinstance(insert_stmt, pgast.InsertStmt)
 
     typeref = ir_stmt.subject.typeref
     if typeref.material_type is not None:
         typeref = typeref.material_type
 
-    type_val = pgast.TypeCast(
-        arg=pgast.StringConstant(val=str(typeref.id)),
-        type_name=pgast.TypeName(name=('uuid',)),
-    )
-    values.append(
-        pgast.ResTarget(name='__type__', val=type_val)
-    )
-
```

Comment:
This is not needed anymore because we don't store __type__ in the backing table anymore, right?

But what if a statement after `Insert` needs __type__? Something like:

```
select (Insert Movie { ... })[is Content]
```

Reply:
Oh yeah that was just some git screwup. All of that is already merged. That's right, for why we aren't doing it.
It'll still get injected into the RETURNING clause of the insert when needed.

(This comes up a lot actually, when injecting type names for the CLI)